api/src/main/java/io/druid/data/input/impl/FileIteratingFirehose.java

```diff
@@ -44,13 +47,26 @@ public FileIteratingFirehose(
   {
     this.lineIterators = lineIterators;
     this.parser = parser;
+    final ParseSpec parseSpec = parser.getParseSpec();
+    if (parseSpec instanceof CSVParseSpec) {
```

Comment:
I was instead thinking that it would be up to the user to avoid using options like `skipHeaderRows` or `hasHeaderRow` in situations where they wouldn't make sense. And if the user did it anyway, they might get weird behavior.

I think if we're ok with that, it should be safe to just have the FileIteratingFirehose call `parser.reset()` whenever it goes to a new file, which would reset the behavior for both `skipHeaderRows` and `hasHeaderRow`. Other scenarios (hadoop indexing, realtime indexing) wouldn't call `parser.reset()` and so they'd just do that behavior on the first line(s) they read. Which is weird, but, I was thinking that would be fine since the user wasn't supposed to configure that in the first place…

Were you wanting to make the behavior smarter somehow?